### PR TITLE
Fix MLTQuery handling of custom term frequencies

### LINKED ISSUE #40125: MLT queries with custom term frequencies
**Elasticsearch version** (`bin/elasticsearch --version`): latest master snapshot, affects all elasticsearch versions with lucene 7.0 or later

**Plugins installed**: []

**JVM version** (`java -version`): openjdk version "11.0.2" 2019-01-15

**OS version** (`uname -a` if on a Unix-like system): Darwin Kernel Version 18.2.0

**Description of the problem including expected versus actual behavior**: 
When using an analyzer with custom term frequencies with MLT-queries:
Expected behavior: Like-text term frequencies would be defined by the analyzer.
Actual behavior: Like-text terms frequencies are defined by number of token occurrences (which omits the custom frequencies).

This has the following implications:
(1) Hit scoring and thus ordering might be incorrect as the input frequencies are not defined by the analyzer.
(2) Parameter `min_term_freq` may work unexpectedly resulting to missing hits as some input terms might be omitted.

See https://issues.apache.org/jira/browse/LUCENE-7854

**Steps to reproduce**:

Added a unit test for the issue in this commit https://github.com/ollik1/elasticsearch/commit/886feeeec0ed4589c7e815390d1ee36aae72f5ae which reproduces the issue . The commit also includes a proposal how to fix it. If you agree it makes sense, I can create a PR based on the commit.

### PR DESCRIPTION
**Pull request by benwtrent**
This applies the fix present in Lucene in our XMoreLikeThisQuery object.

closes: https://github.com/elastic/elasticsearch/issues/40125

related Lucene fix: https://issues.apache.org/jira/browse/LUCENE-8756

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/common/lucene/search/morelikethis/XMoreLikeThisTests.java [hunk 1]
```diff
@@ -8,34 +8,136 @@
...
+            .stream()
+            .map(BooleanClause::getQuery)
+            .map(BoostQuery.class::cast)
+            .filter(x -> ((TermQuery) x.getQuery()).getTerm().text().equals("lucene"))
+            .mapToDouble(BoostQuery::getBoost)
+            .sum();
+
+        // mlt should use the custom frequencies provided by the analyzer so "lucene|10" should be
+        // boosted more than "lucene|1"
+        assertTrue(String.format(Locale.ROOT, "%s should be grater than %s", boost10, boost1), boost10 > boost1);
```
Comment (pmpailis) [inline server/src/test/java/org/elasticsearch/common/lucene/search/morelikethis/XMoreLikeThisTests.java:135]: nit: type in error message `grater` -> `greater`

File: server/src/test/java/org/elasticsearch/common/lucene/search/morelikethis/XMoreLikeThisTests.java [hunk 2]
```diff
@@ -8,34 +8,136 @@
...
+        analyzer.close();
+        reader.close();
```
Comment (pmpailis) [inline server/src/test/java/org/elasticsearch/common/lucene/search/morelikethis/XMoreLikeThisTests.java:137]: nit: could convert to try with resources to ensure these will close gracefully even if the assertion fails

**Review by pmpailis — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/common/lucene/search/morelikethis/XMoreLikeThisTests.java (hunk 1 above)
Comment (benwtrent) [inline server/src/test/java/org/elasticsearch/common/lucene/search/morelikethis/XMoreLikeThisTests.java:135]:
```suggestion
        assertTrue(String.format(Locale.ROOT, "%s should be greater than %s", boost10, boost1), boost10 > boost1);
```

I basically just copied from Lucene, so we also inherited spelling mistakes :)

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**